plugin/pkg/scheduler/schedulercache/node_info_test.go

```diff
@@ -0,0 +1,253 @@
+	}
+}
+
+func TestDiskPressureCondition(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	nodeinfo.diskPressureCondition = v1.ConditionTrue
+	if !reflect.DeepEqual(nodeinfo.DiskPressureCondition(), nodeinfo.diskPressureCondition) {
+		t.Fatalf("get nodeinfo diskPressureCondition error,expected: %v,got: %v", nodeinfo.diskPressureCondition, nodeinfo.DiskPressureCondition())
+	}
+}
+
+func TestNonZeroRequest(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	nodeinfo.nonzeroRequest = &testresource
+	if !reflect.DeepEqual(nodeinfo.NonZeroRequest(), *nodeinfo.nonzeroRequest) {
+		t.Fatalf("get nodeinfo diskPressureCondition error,expected: %v,got: %v", *nodeinfo.nonzeroRequest, nodeinfo.NonZeroRequest())
+	}
+}
+func TestAllocatableResource(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	nodeinfo.allocatableResource = &testresource
+	if !reflect.DeepEqual(nodeinfo.AllocatableResource(), *nodeinfo.allocatableResource) {
+		t.Fatalf("get nodeinfo allocatableResource error,expected: %v,got: %v", *nodeinfo.allocatableResource, nodeinfo.AllocatableResource())
+	}
+}
+
+func TestClone(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	niclone := nodeinfo.Clone()
+	if !reflect.DeepEqual(*niclone, nodeinfo) {
+		t.Fatalf("nodeinfo clone error,expected: %v,got: %v", nodeinfo, *niclone)
+	}
+}
+
+func TesthasPodAffinityConstraints(t *testing.T) {
+	podwithaffinity := makepod("foo", "", &affnity)
+	podwithoutaffinity := makepod("foo", "", &v1.Affinity{})
+	if !reflect.DeepEqual(hasPodAffinityConstraints(podwithaffinity), true) {
+		t.Fatalf("nodeinfo check pod has podaffinityconstraints error,expected: %v,got: %v", true, hasPodAffinityConstraints(podwithaffinity))
+	}
+	if !reflect.DeepEqual(hasPodAffinityConstraints(podwithoutaffinity), false) {
+		t.Fatalf("nodeinfo check pod has no podaffinityconstraints error,expected: %v,got: %v", false, hasPodAffinityConstraints(podwithoutaffinity))
+	}
+}
+
+func TestaddPod(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	podwithaffinity := makepod("foo", "", &affnity)
+	nodeinfo.addPod(podwithaffinity)
+	if !reflect.DeepEqual(nodeinfo.podsWithAffinity, []*v1.Pod{podwithaffinity}) {
+		t.Fatalf("nodeinfo add pod with affinity error,expected: %v,got: %v", []*v1.Pod{podwithaffinity}, nodeinfo.podsWithAffinity)
+	}
+	if !reflect.DeepEqual(nodeinfo.pods, []*v1.Pod{podwithaffinity}) {
+		t.Fatalf("nodeinfo add pod error,expected: %v,got: %v", []*v1.Pod{podwithaffinity}, nodeinfo.pods)
+	}
+
+}
+
+func TestremovePod(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	podwithaffinity := makepod("foo", "", &affnity)
+	podwithoutaffinity := makepod("foo", "", &v1.Affinity{})
+	nodeinfo.addPod(podwithaffinity)
+	nodeinfo.addPod(podwithoutaffinity)

```

Comment:
Let's test exported method firstly; if key logic is not covered, then test the un-exported methods.